ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts

```diff
@@ -173,22 +174,24 @@ export async function reportExceptions(
 ) {
   context.log(`ReportStream response errors: ${response.errorCount}`);
   context.log(`ReportStream response warnings: ${response.warningCount}`);
-  const payloads: SimpleReportReportStreamResponse[] = response.warnings
-    .map(({ id, details }) => ({
-      testEventInternalId: id,
-      isError: false,
-      details,
-    }))
-    .concat(
-      response.errors.map(({ id, details }) => ({
-        testEventInternalId: id,
-        isError: true,
-        details,
-      }))
-    );
+  const payloads: SimpleReportReportStreamResponse[] = response.warnings.flatMap(w => responsesFrom(context, w, false))
+      .concat(response.errors.flatMap(e => responsesFrom(context, e, true)));
   return Promise.all(
     payloads.map((p) =>
       queueClient.sendMessage(Buffer.from(JSON.stringify(p)).toString("base64"))
     )
   );
 }
+
+const responsesFrom = function(context: Context, err: ReportStreamError, isError: boolean):SimpleReportReportStreamResponse[] {
+  if (err.trackingIds) {
+    return err.trackingIds.map(id => ({
+      testEventInternalId: id,
```

Comment:
Is there anything else we or ReportStream need to do to make sure the `trackingElement` is the internal ID of the `TestEvent`?

Reply:
Nvm, I remember we looked at this 👍